packages/router/src/adapters/subgraph/index.ts

```diff
@@ -39,10 +38,10 @@ export const getSdks = (): Record<number, Sdk> => {
 };
 
 export const subgraphContractReader = (): ContractReader => {
-  const { config } = getContext();
+  const { config, logger } = getContext();
   Object.entries(config.chainConfig).forEach(([chainId, { subgraph }]) => {
-    const client = new GraphQLClient(subgraph);
-    sdks[parseInt(chainId)] = getSdk(client);
+    const chainIdNumber = parseInt(chainId);
+    sdks[chainIdNumber] = getSdk(new FallbackSubgraph(logger, chainIdNumber, subgraph));
```

Comment:
The `getSdk` function is generated by the `codegen` command, using the graph schema and the queries outlined in `queries.ts`. It won't be something that's easy to meddle with. [Here's](https://github.com/connext/nxtp/blob/8a05946b82bfd80fdd174a192cfdfeed910ea614/packages/sdk/src/subgraph/subgraph.ts#L207) [some](https://github.com/connext/nxtp/blob/8a05946b82bfd80fdd174a192cfdfeed910ea614/packages/sdk/src/subgraph/subgraph.ts#L325-L329) [examples](https://github.com/connext/nxtp/blob/8a05946b82bfd80fdd174a192cfdfeed910ea614/packages/router/src/adapters/subgraph/subgraph.ts#L347-L349) of how the clients are used.

It would be a good idea to explicitly make the `FallbackSubgraph` extend the `GraphQLClient` class, but I think this is likely the best way to go about it

Reply:
This *is* the best way to go about it? The problem has particularly to do with the `GetBlockNumberQuery` being locked up in the `graphqlsdk.ts` file. If I try to make the get block number call to subgraph internal to the fallback provider, I end up with an antipattern of sorts

Reply:
> This _is_ the best way to go about it?
This sounded really sassy but I actually was just trying to clarify loll